### ISSUE TITLE
successCommentCondition not able to check PR labels

### ISSUE THREAD

**Issue by shahzebrizvi16** [semantic-release/github#920]
I am trying to limit the success comments to only PRs which have not been released on a specific channel already. I am doing this by setting the successCommentCondition to check if a PR has the respective "released on @\<channel>" label already or not. If it does then success commenting should be skipped.

In a nutshell, I am trying to avoid this on PRs:

![image](https://github.com/user-attachments/assets/fd1283a2-f40f-42bd-b7c4-b219df5bbdc3)

I tried doing it [the way specified in the README.md file](https://github.com/semantic-release/github/blob/master/README.md?plain=1#L205) but after testing it I realized that it might be incorrect. This is because within the [Pull Request GitHub API response](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request), the "labels" key is an array of objects which is why the loadash method `-.includes` would not work there.

So instead I tried to set the successCommentCondition this way using the loadash method `-.some` after locally testing with a sample API response:
```
    [
      "@semantic-release/github",
      {
        successCommentCondition:
          "<% return !issue.labels?.some((label) => { return label.name === ('released on @beta'); }) %>"
      }
    ],
```
The snippet above still did not work. My PRs with the label `released on @beta` are still getting commented on if I back-merge the main branch tags to the beta branch.

Can I please get some guidance on this, is it my syntax or is this potentially a bug with the plugin?

Thanks!

**Comment by babblebey** [comment]
Yea @shahzebrizvi16

Thanks for reporting this issue, but we sort of deviated from that format to making the label into a simple array with just their names, meaning we've got 
```
 "labels": ["bug"]
```
...instead of..
```
 "labels": [
    {
      "id": 208045946,
      "node_id": "MDU6TGFiZWwyMDgwNDU5NDY=",
      "url": "https://api.github.com/repos/octocat/Hello-World/labels/bug",
      "name": "bug",
      "description": "Something isn't working",
      "color": "f29513",
      "default": true
    }
  ]
```
and this like didn't exactly make it look like it is in the [Pull Request GitHub API response](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request) truthfully 😞. I did that 

So, if you were to have used what was on the documentation example, that would have worked, hence if your config was 

```
"<% return !issue.labels?includes('released on @beta')) %>"
```

Yea, that would've worked!

But considering the fact that we had stated in the same documentation example that you should use the [Pull Request GitHub API response](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request), so we really are supposed to model exactly after that.

So, we will fix that for sure 👍

**Comment by shahzebrizvi16** [comment]
Also the reason I started looking in to the alternative loadash method is because the original suggestion did not work for me. I just tried it again and it is still commenting on my PR.

Here is how I have it on the release configuration now:
<img width="722" alt="image" src="https://github.com/user-attachments/assets/029aa8cb-8483-4026-b882-74f94a49fc27">

Here it is commenting on the PR:
<img width="972" alt="image" src="https://github.com/user-attachments/assets/53852750-c84b-4342-b365-863e9a9eb94f">

I am wondering if has to do with the way the labels were being manipulated.
